lib/main.js

```diff
@@ -47,6 +47,30 @@ const Hydrogen = {
 
     this.markerBubbleMap = {};
 
+    let skipLanguageMappingsChange = false;
+    store.subscriptions.add(
+      atom.config.onDidChange(
+        "Hydrogen.languageMappings",
+        ({ newValue, oldValue }) => {
+          if (skipLanguageMappingsChange) {
+            skipLanguageMappingsChange = false;
+            return;
+          }
+
+          if (store.runningKernels.size != 0) {
+            skipLanguageMappingsChange = true;
+
+            atom.config.set("Hydrogen.languageMappings", oldValue);
```

Comment:
We may be able to use [`atom.config.transact`](https://atom.io/docs/api/v1.16.0/Config#instance-transact)  here to prevent calling `onDidChange` again which, would be a bit clearer.